internal/controller/ols_app_redis_assets.go

```diff
@@ -101,3 +200,36 @@ func (r *OLSConfigReconciler) generateRedisService(cr *olsv1alpha1.OLSConfig) (*
 
 	return &service, nil
 }
+
+func (r *OLSConfigReconciler) generateRedisSecret(cr *olsv1alpha1.OLSConfig) (*corev1.Secret, error) {
+	randomPassword := make([]byte, 12)
+	_, err := rand.Read(randomPassword)
+	if err != nil {
+		return nil, fmt.Errorf("Error generating random password: %w", err)
+	}
+	// Encode the password to base64
+	encodedPassword := base64.StdEncoding.EncodeToString(randomPassword)
+	passwordHash, err := hashBytes([]byte(encodedPassword))
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate OLS redis password hash %w", err)
+	}
+	secret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Spec.OLSConfig.ConversationCache.Redis.CredentialsSecretRef.Name,
```

Comment:
CredentialsSecretRef is optional , this could be nil ? Also dont see where the default value for the secret is specified (`lightspeed-redis-secret`).  @vishnuchalla

Reply:
Found the Secret name initialization : https://github.com/openshift/lightspeed-operator/pull/25/files#r1522096742